src/js/components/ToggleButtonGroup/__tests__/ToggleButtonGroup-test.tsx

```diff
@@ -0,0 +1,490 @@
+    render(
+      <Grommet>
+        <ToggleButtonGroup options={['one', 'two', 'three']} />
+      </Grommet>,
+    );
+
+    // click on one to make active
+    fireEvent.click(screen.getByRole('radio', { name: 'one' }));
+
+    const activeToggleButtonOne = screen.getByRole('radio', {
+      name: /one/i,
+    });
+
+    // Assert that the aria-checked attribute of the active toggle button is true
+    expect(activeToggleButtonOne).toHaveAttribute('aria-checked', 'true');
+    // click on one to clear value
+    fireEvent.click(screen.getByRole('radio', { name: 'one' }));
+    // // Assert that the aria-checked attribute of the active toggle button is true
+    expect(activeToggleButtonOne).toHaveAttribute('aria-checked', 'false');
+  });
+
+  test('Should call onChange in uncontrolled scenarios', () => {
+    const onChange = jest.fn();
+    const { container } = render(
+      <Grommet>
+        <ToggleButtonGroup
+          options={['one', 'two', 'three']}
+          onChange={onChange}
+        />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+
+    // Simulate a click event on the radio button
+    fireEvent.click(screen.getByRole('radio', { name: 'two' }));
+    expect(onChange).toHaveBeenCalledWith('two');
+  });
+
+  test('Should move to next button when right/down arrow key is pressed and loop', async () => {
+    const user = userEvent.setup();
+    render(
+      <Grommet>
+        <ToggleButtonGroup options={['one', 'two', 'three']} />
+      </Grommet>,
+    );
+
+    const toggleButtonOne = screen.getByRole('radio', {
+      name: /one/i,
+    });
+    const toggleButtonTwo = screen.getByRole('radio', {
+      name: /two/i,
+    });
+
+    await user.tab();
+    await user.type(toggleButtonOne, '{arrowDown}');
+    await user.type(toggleButtonTwo, '{arrowRight}');
+    await user.type(toggleButtonTwo, '{enter}');
+    expect(toggleButtonTwo).toHaveAttribute('aria-checked', 'true');
+  });
+
+  test('Should move to previous button left/up arrow key is pressed and loop', async () => {
+    const user = userEvent.setup();
+    render(
+      <Grommet>
+        <ToggleButtonGroup options={['one', 'two', 'three']} />
+      </Grommet>,
+    );
+
+    const toggleButtonOne = screen.getByRole('radio', {
+      name: /one/i,
+    });
+    const toggleButtonTwo = screen.getByRole('radio', {
+      name: /two/i,
+    });
+
+    await user.tab();
+    await user.type(toggleButtonOne, '{arrowLeft}');
+    await user.type(toggleButtonTwo, '{arrowUp}');
+    await user.type(toggleButtonTwo, '{enter}');
+    expect(toggleButtonTwo).toHaveAttribute('aria-checked', 'true');
+  });
+
+  test('icon with values', () => {
+    const { container } = render(
+      <Grommet>
+        <ToggleButtonGroup
+          options={[
+            { icon: <Add />, value: 'one' },
+            { icon: <Add />, value: 'two' },
+          ]}
+          defaultValue="one"
+        />
+      </Grommet>,
+    );
+
```

Comment:
Can we add an expect here that the button with value "one" should have aria-checked?

Reply:
added